src/main/java/com/dongtronic/diabot/logic/nutrition/NutritionixCommunicator.kt

```diff
@@ -4,46 +4,49 @@ import com.dongtronic.diabot.data.nutrition.NutritionRequestDTO
 import com.dongtronic.diabot.data.nutrition.NutritionResponseDTO
 import com.dongtronic.diabot.exceptions.RequestStatusException
 import com.dongtronic.diabot.util.logger
-import com.fasterxml.jackson.databind.ObjectMapper
-import org.apache.http.client.methods.RequestBuilder
-import org.apache.http.entity.StringEntity
-import org.apache.http.impl.client.HttpClients
-import org.apache.http.util.EntityUtils
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+import reactor.core.publisher.Mono
 
 object NutritionixCommunicator {
     private val appid = System.getenv("nutritionixappid")
     private val secret = System.getenv("nutritionixsecret")
     private const val baseUrl = "https://trackapi.nutritionix.com/v2"
-    private val mapper = ObjectMapper()
+    private val mapper = jacksonObjectMapper()
+    private val client = OkHttpClient()
     private val logger = logger()
 
-    public fun getNutritionInfo(input: String): NutritionResponseDTO {
+    fun getNutritionInfo(input: String): Mono<NutritionResponseDTO> {
         val url = "$baseUrl/natural/nutrients"
 
-        val requestObject = NutritionRequestDTO(input)
+        return Mono.fromCallable {
+            val requestObject = NutritionRequestDTO(input)
 
-        val jsonRequest = mapper.writeValueAsString(requestObject)
+            val jsonRequest = mapper.writeValueAsString(requestObject)
 
-        val client = HttpClients.createDefault()
-        val request = RequestBuilder.post()
+            val mediaType = "application/json".toMediaType()
```

Comment:
Is there no constant for this?

Reply:
No, unfortunately there is not a constant but I can move the definition of `mediaType` outside of the function if that would work better?